driver/src/main/scala/api/commands/writeresult.scala

```diff
@@ -32,8 +32,7 @@ sealed trait WriteResult {
 object WriteResult {
   def lastError(result: WriteResult): Option[LastError] = result match {
```

Comment:
> the question is whether it should really fails, whereas MongoDB indicates ok.

If we decide that the Future should not fail then it would be a breaking change in semantic I guess. Could breaks applications that don't notice the change.

I will try to add a test to check failure behavior of bulk inserts. Could take me a bit of time to figure-out how/where so I'd appreciate any pointer.